frontend/src/pages/TaskMapPage.tsx

```diff
@@ -68,47 +69,77 @@ export const TaskMapPage = () => {
-    targetPosition: Position.Left,
-    draggable: false,
-    position: { x: 550 * idx, y: 80 },
-    data: {
-      label: (
-        <TaskGroup
-          listId={idx}
-          taskIds={synergies.sort((a, b) => a - b)} // FIXME: ordering prevents render bugs
-          selectedTask={selectedTask}
-          setSelectedTask={setSelectedTask}
-          allDependencies={taskRelations.flatMap(
-            ({ dependencies }) => dependencies,
-          )}
-          disableDragging={disableDrag}
-        />
-      ),
-    },
-  }));
-
-  const edges = taskRelations.flatMap(({ dependencies }, idx) =>
-    dependencies.map(({ from, to }) => {
-      const targetGroupIdx = taskRelations.findIndex(({ synergies }) =>
-        synergies.includes(to),
-      );
+  useEffect(() => {
+    if (!divRef || taskRelations.length === 0) return;
+
+    const measuredRelations = taskRelations.map((relation, idx) => {
+      // calculate taskgroup height
+      let height = 40;
+
+      relation.synergies.forEach((taskId) => {
+        const task = tasks.find(({ id }) => id === taskId);
+        if (!task) return;
+        height += 40;
+        // calculate text height
+        divRef.textContent = task.name;
+        height += divRef.offsetHeight;
+        divRef.textContent = '';
+      });
+      return { id: `${idx}`, width: 436, height, ...relation };
+    });
 
+    const graph = getAutolayout(measuredRelations);
+
+    const groups = measuredRelations.map(({ id, synergies }) => {
+      const node = graph.node(id);
       return {
-        id: `from-${from}-to-${to}`,
-        source: String(idx),
-        sourceHandle: `from-${from}`,
-        target: String(targetGroupIdx),
-        targetHandle: `to-${to}`,
-        type: 'custom',
-        data: { disableInteraction: disableDrag },
+        id,
+        type: 'special',
+        sourcePosition: Position.Right,
+        targetPosition: Position.Left,
+        draggable: false,
+        // dagre coordinates are in the center, calculate top left corner
+        position: { x: node.x - node.width / 2, y: node.y - node.height / 2 },
+        data: {
+          label: (
+            <TaskGroup
+              listId={id}
+              taskIds={synergies}
+              selectedTask={selectedTask}
+              setSelectedTask={setSelectedTask}
+              allDependencies={taskRelations.flatMap(
+                ({ dependencies }) => dependencies,
+              )}
+              disableDragging={disableDrag}
+            />
+          ),
+        },
       };
-    }),
-  );
+    });
+
+    const edges = measuredRelations.flatMap(({ dependencies }, idx) =>
+      dependencies.map(({ from, to }) => {
+        const targetGroup = measuredRelations.find(({ synergies }) =>
+          synergies.includes(to),
+        );
+
+        return {
+          id: `from-${from}-to-${to}`,
+          source: String(idx),
+          sourceHandle: `from-${from}`,
+          target: String(targetGroup!.id),
```

Comment:
```suggestion
          target: targetGroup!.id,
```
One extra conversion, that I missed the first time